src/pyspiffe/workloadapi/default_workload_api_client.py

```diff
@@ -0,0 +1,138 @@
+from pyspiffe.workloadapi.workload_api_client import WorkloadApiClient
+from pyspiffe.internal.defaults import Defaults
+
+class DefaultWorkloadApiClient(WorkloadApiClient):
+    """
+    Default implementation for a SPIFFE Workload API Client.
+    """
+
+    # TODO: Not sure we need all the arguments, see newClient()
+    def __init__(self, spiffe_socket_path=None, exponential_backoff_policy=None,
+                 executor_service=None):
+
+        if spiffe_socket_path == None:
+            self.spiffe_socket_path=Defaults.spiffe_socket_path
+        else:
+            self.spiffe_socket_path=spiffe_socket_path
+
+        # TODO defaults for these, or delete them entirely
+        self.exponential_backoff_policy=exponential_backoff_policy
+        self.executor_service=executor_service
+
+
+    # TODO: Is this pythonic? Any benefit of this being a factory object vs
+    # just using the constructor?
+    @classmethod
```

Comment:
I guess I need to leave this to someone with more recent Python experience. Does Python not allow overriding contructors? Seems like I've done that before, but maybe not.

Since this is pretty fundamental to the design of the API I do think this is something we should decide now. I.e. if we release the first version of the library using a constructor, but then later switch to a factor pattern that would change the API which would be a bad idea.

Reply:
No, it does not allow allow for overriding constructors .. this is when we would want another method to instantiate an object of the class.... Agree with you that changing the API would be a bad idea but I don't see how changing to a factor pattern here would change the API. As far as I am concerned this workload_api_client should not be visible for the clients of py-spiffe. And I am not a fan of doing thing for a future we don't know if will ever come. With that said, I don't think this is a fundamental problem or issue. I have my preference but I could go with either way. So if you feel strongly about keeping it, I am fine.

Reply:
The Workload API client _is_ part of the API in the other two libraries, because although the library users will generally prefer to use the X509Source they may sometimes want to make a one-shot call to fetch an SVID, and that functionality is provided by the workload API client.

Reply:
Hmmmm very good point @maxlambrecht, I was not considering this usage. Thanks!

Reply:
For now I've removed the factory method and left only the constructor. I continue to have no opinion over which is "the right way". It is easier to add one later than take it away, so that's why it's removed for now.